### ISSUE TITLE
[2024.03] PHP Warnings

### ISSUE THREAD

**Issue by tobiasd** [friendica/friendica#13761]
Please report here any PHP Warning message obtained running the 2023.12 stable or 2024.03-dev/rc versions.

Replaces #13157

**Comment by C0RD** [comment]
`PHP Warning:  Undefined property: stdClass::$value in /var/www/friendica.cord.de/addon/bluesky/bluesky.php on line 1324`

```
worker [WARNING]: E_WARNING: Undefined property: stdClass::$value {"code":2,"message":"Undefined property: stdClass::$value","file":"/var/www/friendica.cord.de/addon/bluesky/bluesky.php","line":1324} - {"file":"bluesky.php","line":1324,"function":"bluesky_add_media","request-id":"658a47b2e7713","stack":"ErrorHandler::handleError (1324), bluesky_add_media (1130), bluesky_process_post (1093), bluesky_fetch_feed (14), bluesky_feed_run (581), Worker::execFunction (428), Worker::execute (121), Worker::processQueue (87)","uid":"98ae67","process_id":1455906}
2023-12-26T03:25:58Z worker [ERROR]: Uncaught Exception TypeError: "bluesky_get_content(): Argument #2 ($record) must be of type stdClass, null given, called in /var/www/friendica.cord.de/addon/bluesky/bluesky.php on line 1324" at /var/www/friendica.cord.de/addon/bluesky/bluesky.php line 1175 {"exception":"TypeError: bluesky_get_content(): Argument #2 ($record) must be of type stdClass, null given, called in /var/www/friendica.cord.de/addon/bluesky/bluesky.php on line 1324 and defined in /var/www/friendica.cord.de/addon/bluesky/bluesky.php:1175\nStack trace:\n#0 /var/www/friendica.cord.de/addon/bluesky/bluesky.php(1324): bluesky_get_content()\n#1 /var/www/friendica.cord.de/addon/bluesky/bluesky.php(1130): bluesky_add_media()\n#2 /var/www/friendica.cord.de/addon/bluesky/bluesky.php(1093): bluesky_process_post()\n#3 /var/www/friendica.cord.de/addon/bluesky/bluesky_feed.php(14): bluesky_fetch_feed()\n#4 /var/www/friendica.cord.de/src/Core/Worker.php(581): bluesky_feed_run()\n#5 /var/www/friendica.cord.de/src/Core/Worker.php(428): Friendica\\Core\\Worker::execFunction()\n#6 /var/www/friendica.cord.de/src/Core/Worker.php(121): Friendica\\Core\\Worker::execute()\n#7 /var/www/friendica.cord.de/bin/worker.php(87): Friendica\\Core\\Worker::processQueue()\n#8 {main}"} - {"file":null,"line":null,"function":null,"request-id":"658a47b2e7713","stack":"ErrorHandler::handleException (128), ErrorHandler::Friendica\\Core\\Logger\\Handler\\{closure}","uid":"98ae67","process_id":1455906}

```

**Comment by C0RD** [comment]
libpng warning: Interlace handling should be turned on when using png_read_image

**Comment by AlfredSK** [comment]
2024.03-dev, PHP 8.1

```
Undefined array key 19 in .../src/Module/Conversation/Community.php on line 136
Trying to access array offset on value of type null in .../src/Module/Conversation/Community.php on line 136
Undefined array key 0 in .../src/Module/Conversation/Community.php on line 135
Trying to access array offset on value of type null in .../src/Module/Conversation/Community.php on line 135
Undefined array key 29 in .../src/Module/Conversation/Community.php on line 136
Trying to access array offset on value of type null in .../src/Module/Conversation/Community.php on line 136
```
Addressed in https://github.com/friendica/friendica/pull/13825

**Comment by dmorley** [comment]
composer install/update stuck, this happened years back but I can not resolve again.

  Problem 1
    - fxp/composer-asset-plugin is locked to version v1.4.6 and an update of this package was not requested.
    - fxp/composer-asset-plugin v1.4.6 requires composer-plugin-api ^1.0 -> found composer-plugin-api[2.6.0] but it does not match the constraint.

**Comment by MrPetovan** [comment]
@dmorley Please use the provided Composer binary bundled with the project in `bin/composer.phar`. A typical command from Friendica base directory looks like this:
```bash
bin/composer.phar install --no-dev
```

**Comment by dmorley** [comment]
> @dmorley Please use the provided Composer binary bundled with the project in `bin/composer.phar`. A typical command from Friendica base directory looks like this:
> 
> ```shell
> bin/composer.phar install --no-dev
> ```

Yes this is what I am using and kicks the error.

**Comment by dmorley** [comment]
Ah, that must have been messed up, checking it out again from origin/develop got me going, thanks!

**Comment by AlfredSK** [comment]
Possibly related to #13900 

```unpack(): Type A: not enough input, need 4, have 0 in .../src/Object/Image.php on line 118```

Full error message:
```
app [WARNING]: E_WARNING: unpack(): Type A: not enough input, need 4, have 0 {"code":2,"message":"unpack(): Type A: not enough input, need 4, have 0","file":".../src/Object/Image.php","line":118} - {"file":null,"line":null,"function":"unpack","request-id":"65d088dbc64d0","stack":"ErrorHandler::handleError, unpack (118), Image::isAnimatedWebP (107), Image::useImagick (78), Image::__construct (399), Photo::getPhotoById (133), Photo::rawContent (244), BaseModule::run (109), BaseApi::run (711), App::runFrontend (55)","uid":"84d8ec","process_id":3383557}
```

2024.03-dev, PHP 8.1

Addressed in #13904

**Comment by AlfredSK** [comment]
Also possibly related to #13900 

```preg_match_all(): Null byte in regex in .../src/Object/Image.php on line 103```

Full error message:
```
worker [WARNING]: E_WARNING: preg_match_all(): Null byte in regex {"code":2,"message":"preg_match_all(): Null byte in regex","file":".../src/Object/Image.php","line":103} - {"file":null,"line":null,"function":"preg_match_all","request-id":"65d08c0d2e583","stack":"ErrorHandler::handleError, preg_match_all (103), Image::useImagick (78), Image::__construct (147), Link::fetchMimeType (79), Link::getByLink (144), Processor::replaceEmojis (862), Processor::processContent (488), Processor::createItem (782), Receiver::routeActivities (239), Queue::process (347), Queue::processReplyByUri (1193), Processor::postItem (783), Receiver::routeActivities (239), Queue::process (347)","uid":"792662","process_id":3554350}
```

2024.03-dev, PHP 8.1

Addressed in #13904 and https://github.com/friendica/friendica/pull/13908

**Comment by ghost** [comment]
geeting this since i switched to php 8.3 an RC

PHP Warning: Trying to access array offset on null in /view/smarty3/compiled/8a/7b/70/8a7b70de132de62462cc95bf286b4d9aabd73698_2.file.threaded_conversation.tpl.php on line 46'

**Comment by ne20002** [comment]
Since 2024.03-rc in the cron container:
`Deprecated: mb_strlen(): Passing null to parameter #1 ($string) of type string is deprecated in /var/www/html/src/Model/Post/Counts.php on line 55`

Addressed in https://github.com/friendica/friendica/pull/13970

**Comment by AlfredSK** [comment]
Undefined array key "author-alias" in .../mod/photos.php on line 1219

```
app [WARNING]: E_WARNING: Undefined array key "author-alias" {"code":2,"message":"Undefined array key \"author-alias\"","file":".../mod/photos.php","line":1219} - {"file":"photos.php","line":1219,"function":"photos_content","request-id":"65e803fdac01f","stack":"ErrorHandler::handleError (1219), photos_content (96), LegacyModule::runModuleFunction (73), LegacyModule::content (250), BaseModule::run (711), App::runFrontend (55)","uid":"d1472f","process_id":2470535}
```

2024.03-rc, PHP 8.1

Addressed in https://github.com/friendica/friendica/pull/13970

**Comment by AlfredSK** [comment]
```
Undefined variable $success {"code":2,"message":"Undefined variable $success","file":".../src/Protocol/Delivery.php","line":352} - {"file":"Delivery.php","line":352,"function":"deliverDFRN","request-id":"65f4d98d8ee1c","stack":"ErrorHandler::handleError (352), Delivery::deliverDFRN (230), Delivery::deliver (39), BulkDelivery::execute, Worker::execFunction (386), Worker::execute (121), Worker::processQueue (87)","uid":"295bcc","process_id":2426362}

```

2024-03-rc, PHP 8.1

Addressed in #14004
